### PR TITLE
check slash cmd role perms in interaction handler

### LINKED ISSUE #321: Discord permissions v2 no longer supports bulk update of interaction command permissions
## Describe the bug
Previously we configured the permissions for the slash-commands on the 'ready' event. This no longer works due to Discord Permissions v2 api changes such as removing the bulk permission setting endpoint (which is being called in our code).

## Expected behavior
The permission handling should work for all the slash-commands.

The most straight-forward fix would likely be to implement a permissions handler similar to the current text-command permission handler. This would result in all of the slash-commands being accessible to all users (but they won't have permission to use them and will get a "No permission" type response).

Stack Overflow answer suggesting the simple role check in the code:
https://stackoverflow.com/questions/72048570/403-error-when-setting-application-command-permissions-on-discord

Another alternative is for admins to deploy commands with disabled permissions and then manually set command permissions through the Discord UI. But a programmatic solution seems preferable.

## To Reproduce
Start the bot and observe error occurring on ready event:
`
DiscordAPIError: 405: Method Not Allowed
`

## Additional context
Discord API discussing application command permissions:
https://discord.com/developers/docs/interactions/application-commands#permissions

### PR DESCRIPTION
**Pull request by mikejoh12**
## What issue is this solving?

Closes #321

### Description

Removes slash command permission setting by bulk update method as it's no longer supported by Discord in permissions v2. Moves role permission checks into a handler which checks permissions on interactionCreate event. Removes the default permissions false setting for the slash commands. Guild members will be able to launch all slash commands but a reply error message is presented for commands where the user does not have permission (similar to current text commands).

## Any helpful knowledge/context for the reviewer?

- Is a re-seeding of the database necessary? No
- Any new dependencies to install? No
- Any special requirements to test?
Redeploy the commands with deployCommands.js.
Check Discord Server Settings - Interactions - "Your test bot" to see the current "Discord" permissions for your slash commands. Confirm that your bot slash commands are availabe to @everyone and All channels and are Synced. It's possible you may need to adjust the settings since they were previously limited to some roles. Basically, all users should have access to all slash commands in the Discord settings (since we handle the permissions with the bot).

### Please make sure you've attempted to meet the following coding standards
- [X] Code has been tested and does not produce errors
- [X] Code is readable and formatted
- [X] There isn't any unnecessary commented-out code

### CODE REVIEW

**Review by aedwardg — APPROVED** [review]
This is fine for now. The discord.js PR that got merged with the default perm stuff isn't in the most recent release so we can wait until that is officially released before revisiting that.

**Review by lyallstewart — APPROVED** [review]

File: events/interactionCreate.js [hunk 1]
```diff
@@ -1,8 +1,28 @@
...
+
+function isAuthorized(interaction, commandRoles) {
+  const allowedRoles = commandRoles.get(interaction.commandName);
+  if (
+    allowedRoles[0] === '@everyone' ||
+    interaction.member.roles.cache.some((r) => allowedRoles.includes(r.id))
+  ) {
+    return true;
+  }
+  return false;
```
Comment (Victoria-DR) [inline events/interactionCreate.js:11]: Could just return the statement and not use a conditional here.

File: events/interactionCreate.js [hunk 2]
```diff
@@ -1,8 +1,28 @@
...
+}
+
 module.exports = {
   name: 'interactionCreate',
   async execute(interaction) {
     if (!interaction.isCommand()) return;
 
+    if (!isAuthorized(interaction, commandRoles)) {
+      return await interaction.reply({
+        content: `You don't have permissions for this command`,
```
Comment (Victoria-DR) [inline events/interactionCreate.js:21]: Can change `You don't have permissions for this command` to `You don't have permission to use this command.`.

**Review by Victoria-DR — APPROVED** [review]
LGTM, thanks for looking into and fixing this! I've added two comments but those are for small changes that aren't too important, so I've approved so as not to block merging. I'm assuming this PR will be merged after #320, so before merging this one, you'll need to make another commit removing the `.setDefaultPermission(false)` from the ban and unban commands.

File: events/interactionCreate.js (hunk 1 above)
Comment (mikejoh12) [reply events/interactionCreate.js]: Good catch. I made this small refactor.

File: events/interactionCreate.js (hunk 2 above)
Comment (mikejoh12) [reply events/interactionCreate.js]: Done

**Pull request merged**